### PR TITLE
chore: Suppress messages generated by CodeAnalyzers

### PR DESCRIPTION
**Pull request by filzrev**
This PR aims to resolve some part of #9207 that can be fixed by Visual Studio's code fixer and handled with small code modifications.

Please review If there are changes that contains undesirable code changes.
Then I'll remove that changes.

| ID      |   Commit                                                                        | Description         |
|---------|---------------------------------------------------------------------------------|---------------------|
| CA1018  | https://github.com/dotnet/docfx/commit/51bbfa817dade6de2a7c645676ac7fd1091bc27e | [CA1018: Mark attributes with AttributeUsageAttribute](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1018)
| CA1041  | https://github.com/dotnet/docfx/commit/b614c9ece703a870d69a2f1e47d62d50ec09f1d3 | [CA1041: Provide ObsoleteAttribute message](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1041)
| CA1050  | https://github.com/dotnet/docfx/commit/c0f8515a3e1d6a89f837b1d866aadcf1f52eba7b | [CA1050: Declare types in namespaces](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1050)
| CA1507  | https://github.com/dotnet/docfx/commit/4dc6426de52f016db6b8f6e687e30289dfcf3d4d | [CA1507: Use nameof in place of string](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1507)
| CA1822  | https://github.com/dotnet/docfx/commit/591c6ed9a48e139ec6ad342273dfa21b759417b9 | [CA1822: Mark members as static](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1822)
| CA1825  | https://github.com/dotnet/docfx/commit/80d6c10c0aa15c15339e908db5e515543ba41372 | [CA1825: Avoid zero-length array allocations](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1825)
| CA1827  | https://github.com/dotnet/docfx/commit/a44bc8d9ed267754d820bd25acd5446c3521a7ab | [CA1827: Do not use Count()/LongCount() when Any() can be used](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1827)
| CA1829  | https://github.com/dotnet/docfx/commit/e758c9f620198e2a99510c66bb86a3bf0c08dbbe | [CA1829: Use Length/Count property instead of Enumerable.Count method](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1829)
| CA1830  | https://github.com/dotnet/docfx/commit/027807a722eaa2407e7375053661b4f3388bd4b9 | [CA1830: Prefer strongly-typed Append and Insert method overloads on StringBuilder](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1830)
| CA1834  | https://github.com/dotnet/docfx/commit/254d17291edc004ebcdae53232c713558a689327 | [CA1834: Use StringBuilder.Append(char) for single character strings](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1834)
| CA1836  | https://github.com/dotnet/docfx/commit/cac85d0348835fc52e33ba51db611a17614bc00c | [CA1836: Prefer IsEmpty over Count when available](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1836)
| CA1841  | https://github.com/dotnet/docfx/commit/27c0251cf3c771d25845285b82f30fc6164b91fc | [CA1841: Prefer Dictionary Contains methods](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1841)
| CA1846  | https://github.com/dotnet/docfx/commit/c78479944eaa19117abc8636292e497bedc6a7ad | [CA1846: Prefer AsSpan over Substring](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1846)
| CA1847  | https://github.com/dotnet/docfx/commit/cf3f1d47fd6d2a50a98851d014270eb677c9cf19 | [CA1847: Use string.Contains(char) instead of string.Contains(string) with single characters](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1847)
| CA1850  | https://github.com/dotnet/docfx/commit/12383b50cdc186f01545d475495b4ad1971ffb33 | [CA1850: Prefer static HashData method over ComputeHash](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1850)
| CA2208  | https://github.com/dotnet/docfx/commit/f7e57cd08554f6dc9c39516addf414b08e160d14 | [CA2208: Instantiate argument exceptions correctly](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/CA2208)
| CA2249  | https://github.com/dotnet/docfx/commit/fbbb5a1e32450f380edc3cc6e06758c3482a0694 | [CA2249: Consider using String.Contains instead of String.IndexOf](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2249)

| ID      | Commit                                                                          | Description         |
|---------|---------------------------------------------------------------------------------|---------------------|
| IDE0001 | https://github.com/dotnet/docfx/commit/99e2d8e0d28a28e0da5cbeb32f5c457e033c9eb5 | [Simplify name (IDE0001)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0001)
| IDE0002 | https://github.com/dotnet/docfx/commit/36242f158fc7be89ca076209535db26d078bfbe2 | [Simplify member access (IDE0002)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0002)
| IDE0003 | https://github.com/dotnet/docfx/commit/1df792525c104869b29539442dd138cab9449044 | [this preferences (IDE0003)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0003-ide0009)
| IDE0004 | https://github.com/dotnet/docfx/commit/073f67596a142d88ff798c74a62ee879f8218412 | [Remove unnecessary cast (IDE0004)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0004)
| IDE0005 | https://github.com/dotnet/docfx/commit/0fe71592c6d3b0c187dfbf92a67badb5e275ad74 | [Remove unnecessary using directives (IDE0005)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0005)
| IDE0007/IDE0008 | https://github.com/dotnet/docfx/commit/09181f4e84c9660cfdac9cced1e40562b4eaf645 | ['var' preferences (IDE0007 and IDE0008)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0007-ide0008)
| IDE0017 | https://github.com/dotnet/docfx/commit/27c2b045aba636cc206a0b866760039425ba7e28 | [Use object initializers (IDE0017)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0017)
| IDE0018 | https://github.com/dotnet/docfx/commit/6e57632d299436c9eb00f8bb881c51ca62a10bbf | [Inline variable declaration (IDE0018)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0018)
| IDE0019 | https://github.com/dotnet/docfx/commit/3b86b47f28bbe694a3bc57885ba53fe0942f6ed4 | [Use pattern matching to avoid 'as' followed by a 'null' check (IDE0019)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0019)
| IDE0029 | https://github.com/dotnet/docfx/commit/9dd355c05418f4d34fcbdd846954469787adbc1d | [Null check can be simplified (IDE0029)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0029-ide0030-ide0270)
| IDE0031 | https://github.com/dotnet/docfx/commit/22175e897f1e6482e29eb74320bae368272b2338 | [Use null propagation (IDE0031)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0031)
| IDE0034 | https://github.com/dotnet/docfx/commit/7523e16fc2c5f22cb4bec3ec17a84581137aa1e3 | [Simplify 'default' expression (IDE0034)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0034)
| IDE0036 | https://github.com/dotnet/docfx/commit/65da193d5dedba274dc34798f2d02321af5e7760 | [Order modifiers (IDE0036)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0036)
| IDE0038 | https://github.com/dotnet/docfx/commit/5f19db3a211a913d4b2be36a62e04235c4772568 | [Use pattern matching to avoid 'is' check followed by a cast (IDE0038)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0020-ide0038)
| IDE0039 | https://github.com/dotnet/docfx/commit/8446077eff9c54ba03c15218be9e2d375a9753d2 | [Use local function instead of lambda (IDE0039)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0039)
| IDE0041 | https://github.com/dotnet/docfx/commit/2ce102af31848cbe924d862a6f8f0397062dd670 | [Use 'is null' check (IDE0041)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0041)
| IDE0044 | https://github.com/dotnet/docfx/commit/a661ac1d0a2ae199026dac2b3366cd2c94c82f04 | [Add readonly modifier (IDE0044)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0044)
| IDE0049 | https://github.com/dotnet/docfx/commit/1bf1a9bbd66349ef9089ba5b26ffa18a2c1b290b | [Use language keywords instead of framework type names for type references (IDE0049)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0049)
| IDE0054 | https://github.com/dotnet/docfx/commit/99585f56e0afb38425c8ee54841323978f43fd13 | [Use compound assignment (IDE0054 and IDE0074)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0054-ide0074)
| IDE0058 | https://github.com/dotnet/docfx/commit/a425568b1d089379313714b2f378175228020b23 | [Remove unnecessary expression value (IDE0058)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0058) (For test projects)
| IDE0059 | https://github.com/dotnet/docfx/commit/f43c8621a18cb297ac8b0fd2d5d71b505ca66e8e | [Remove unnecessary value assignment (IDE0059)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0059) (For test projects)
| IDE0062 | https://github.com/dotnet/docfx/commit/3597d7a2c2f48440d9abe1ecd3c8c00b5ee99196 | [Make local function static (IDE0062)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0062)
| IDE0065 | https://github.com/dotnet/docfx/commit/c246e9e587660f50de422b413ff5c43bf7c8c677 | ['using' directive placement (IDE0065)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0065)
| IDE0071 | https://github.com/dotnet/docfx/commit/16686526f2ece55259238eeb70fe4b3ce4c8adc6 | [Simplify interpolation (IDE0071)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0071)
| IDE0079 | https://github.com/dotnet/docfx/commit/bcef7fce811a2a2b090c96d6bd38ec69d6706daa | [Remove unnecessary suppression (IDE0079)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0079)
| IDE0083 | https://github.com/dotnet/docfx/commit/f302cc6970bc15d62ea63b4d91fb9fa4c6e9bd57 | [Use pattern matching (not operator) (IDE0083)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0083)
| IDE0090 | https://github.com/dotnet/docfx/commit/22e4c7a65b922613b369a2e5462ebaef29781cdf | [Simplify new expression (IDE0090)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0090)
| IDE0110 | https://github.com/dotnet/docfx/commit/82511bff6f116b3b3fe8ea3e4898cca2e458bb6f | [Remove unnecessary discard (IDE0110)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0110)
| IDE0161 | https://github.com/dotnet/docfx/commit/e10d1d5e47fbc27b71baa1c3843d0cde498b3334 | [Namespace declaration preferences (IDE0160 and IDE0161)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0160-ide0161)
| IDE0200 | https://github.com/dotnet/docfx/commit/5e3df45818af8a331f3bfe82eb11fa348dc0449b | [Remove unnecessary lambda expression (IDE0200)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0200)
| IDE1006 | https://github.com/dotnet/docfx/commit/584c5ecc89db647e40dd9c2c06a8cda3d5420802 | Naming rules for static/internal/private fields

**TODO:**
Following tasks will be handled  by another PR.
- Suppress other code that require some code modification.
- Change resolved analyzer rules level from suggestion to warning.
- Try to using [more strict analyzer level](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/overview?tabs=net-7#enable-additional-rules) (`Default` -> `Recommended`) 
- Apply C#12 specific code analyzer rules. (It needs be compiled when targeting .NET 6 SDK)

### CODE REVIEW

File: src/Docfx.Plugins/OutputFileCollection.cs [hunk 1]
```diff
@@ -35,5 +35,5 @@ public class OutputFileCollection : ObservableDictionary<string, OutputFileInfo>
 
     #endregion
 
-    public bool ContainsValue(OutputFileInfo value) => Values.Contains(value);
+    public bool ContainsValue(OutputFileInfo value) => ContainsValue(value);
```
Comment (austindrenski) [inline src/Docfx.Plugins/OutputFileCollection.cs:38]: Just lurking and noticed this looking suspiciously recursive (probably a false positive from `CA1841`)

**Review by austindrenski — CHANGES_REQUESTED** [review]

File: src/Docfx.Plugins/OutputFileCollection.cs (hunk 1 above)
Comment (filzrev) [reply src/Docfx.Plugins/OutputFileCollection.cs]: Thanks for pointing that out.
It's my fault for manually editing the code.

I've reverted code and suppress CA1841 message with commit (https://github.com/dotnet/docfx/pull/9224/commits/fef21e68306cc6bf659c9e40403461fd1e1bcc35)

**Comment by filzrev** [comment]
I've added a commit (https://github.com/dotnet/docfx/pull/9224/commits/33554b230ac3d4e8112be08bcd632a6c5b853f5a) to resolve [CA2011](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2011) message.
If this setter property called. It cause StackOverflow exception.

**Review by yufeih — APPROVED** [review]
Thank you @filzrev !

**Pull request merged**